### PR TITLE
 YLP-389 Teams: Leave team actions

### PR DESCRIPTION
**Pull request by yvon-dblg**

### CODE REVIEW

File: packages/yourloops/pages/hcp/team-leave-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    return null;
+  }
+
+  if (onlyMember) {
+    return (
+      <DialogTitle id="team-leave-dialog-title">
+        <strong>{t("team-leave-dialog-and-del-title")}</strong>
+        <br />
+        <span id="team-leave-dialog-title-team-name">{teamName}</span>
+      </DialogTitle>
```
Comment (leovalette) [inline packages/yourloops/pages/hcp/team-leave-dialog.tsx:113]: This code is duplicated 3 times, the only difference is on the translation in the strong balise.
Maybe define a variable for it and return somthing like 
`
<DialogTitle id="team-leave-dialog-title">
        <strong>{t(myVariable}</strong>
        <br />
        <span id="team-leave-dialog-title-team-name">{teamName}</span>
 </DialogTitle>`

File: packages/yourloops/pages/hcp/team-leave-dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        <DialogContentText id="team-leave-dialog-question">
+          <Trans i18nKey="team-leave-dialog-and-del-question" t={t} components={{ strong: <strong /> }} parent={React.Fragment}>
+            Since you are the only &quot;member&quot; in this team, {{ teamName }} will be <strong>permanently deleted</strong> if you leave it.
+          </Trans>
+        </DialogContentText>
+        <DialogContentText id="team-leave-dialog-consequences">
+          {t("team-leave-dialog-and-del-consequences")}
+        </DialogContentText>
+      </DialogContent>
+    );
```
Comment (leovalette) [inline packages/yourloops/pages/hcp/team-leave-dialog.tsx:155]: Same comment as on the previous method

File: packages/yourloops/pages/hcp/team-leave-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+  if (onlyMember) {
+    return (
+      <DialogActions>
+        <Button id="team-leave-dialog-button-cancel" onClick={handleClose} disabled={buttonsDisabled} className={classes.buttonCancel} color="secondary" variant="contained">
+          {t("Cancel")}
+        </Button>
+        <Button id="team-leave-dialog-button-leave" onClick={handleLeaveTeam} disabled={buttonsDisabled} color="secondary" className={classes.buttonLeaveAndDel} variant="contained">
+          {t("team-leave-dialog-button-leave-and-del")}
+        </Button>
+      </DialogActions>
```
Comment (leovalette) [inline packages/yourloops/pages/hcp/team-leave-dialog.tsx:199]: Same, we can probably avoid these duplications

**Review by leovalette — CHANGES_REQUESTED** [review]
General remarks : 
- Some lines are too long, we need to define editor/lint rules about this, otherwise it will generate "noises changes" on others PR.
- Some code is duplicated
- I still don't like using for...of/for over es6 methods like map/reduce/filter 😄

File: packages/yourloops/lib/auth/api.ts [hunk 4]
```diff
@@ -516,6 +516,39 @@ class AuthApi extends EventTarget {
...
+      const thisTeam = this.teams[i];
+      if (thisTeam.id === team.id && Array.isArray(thisTeam.members)) {
+        const userId = this.user.userid;
+        const idx = thisTeam.members.findIndex((tm: TeamMember) => tm.userId === userId);
+        if (idx > -1) {
+          this.teams.splice(i, 1);
+        }
+        break;
+      }
+    }
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/yourloops/lib/auth/api.ts:545]: Temp code used as a dummy API for now.
No need to worry.

[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/yourloops/lib/auth/api.ts:545]: Temp code used as a dummy API for now.
No need to worry.

File: packages/yourloops/pages/hcp/team-card.tsx [hunk 5]
```diff
@@ -152,6 +157,11 @@ function TeamCard(props: TeamCardProps): JSX.Element {
       {t("button-team-add-member")}
     </Button>
   );
+  const buttonLeaveTeam = (
+    <Button id={`team-card-${team.id}-button-leave-team`} className={classes.buttonActionFirstRow} startIcon={<ExitToAppIcon color="primary" />} onClick={handleClickLeaveTeam}>
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-card.tsx:161]: A PR to fix all theses little things on the team screen will follow.

File: packages/yourloops/pages/hcp/team-leave-dialog.tsx [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+  let userIsAdmin = false;
+  for (const member of members) {
+    if (member.role === "admin") {
+      nAdmin++;
+      if (member.userId === userId) {
+        userIsAdmin = true;
+      }
+    }
+  }
+  return userIsAdmin && nAdmin === 1;
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-leave-dialog.tsx:97]: I'm not fan to only use callbacks function for everything.
I have use for...of to not have complains already.

File: packages/yourloops/pages/hcp/team-leave-dialog.tsx (hunk 1 above)
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-leave-dialog.tsx:113]: A PR to fix all theses little things on the team screen will follow.

File: packages/yourloops/pages/hcp/team-leave-dialog.tsx (hunk 2 above)
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-leave-dialog.tsx:155]: A PR to fix all theses little things on the team screen will follow.

File: packages/yourloops/pages/hcp/team-leave-dialog.tsx (hunk 3 above)
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-leave-dialog.tsx:199]: A PR to fix all theses little things on the team screen will follow.

**Review by leovalette — APPROVED** [review]

**Pull request merged**